src/rpc/rawtransaction.cpp

```diff
@@ -650,16 +650,37 @@ static RPCHelpMan combinerawtransaction()
 {
 
     UniValue txs = request.params[0].get_array();
+
+    // Can't merge < 2 items
+    if (txs.size() < 2) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Missing transactions");
+    }
+
     std::vector<CMutableTransaction> txVariants(txs.size());
 
-    for (unsigned int idx = 0; idx < txs.size(); idx++) {
+    for (unsigned int idx {0}; idx < txs.size(); idx++) {
         if (!DecodeHexTx(txVariants[idx], txs[idx].get_str())) {
             throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed for tx %d. Make sure the tx has at least one input.", idx));
         }
     }
 
-    if (txVariants.empty()) {
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Missing transactions");
+    { // Test Tx relation for mergeability. This involves stripping sigscripts from each tx and ensuring txid is consistent.
+        std::vector<CMutableTransaction> txVariantsCopy(txVariants);
+        Txid txid{};
+        for (unsigned int k{0}; k < txVariantsCopy.size(); k++) {
+            // Remove all scriptSigs and scriptWitnesses from inputs
+            for (CTxIn& input : txVariantsCopy[k].vin) {
+                input.scriptSig.clear();
+                input.scriptWitness.SetNull();
```

Comment:
I don't think it's needed to clear the script witness, as it's not taken into account when calculating the txid